### PR TITLE
How to interact with Lisk using viem

### LINKED ISSUE #40: New page: Interacting with the blockchain with viem
### Description

Explain step by step how to connect to the Lisk blockchain via viem.

### Location

Building on Lisk / Interacting with the blockchain / with viem

### Additional Information
https://docs.base.org/tools/viem

### PR DESCRIPTION
**Pull request by Tschakki**
### What was the problem?

This PR resolves #40

### CODE REVIEW

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+ 
+export const publicClient = createPublicClient({
+  chain: liskSepolia,
+  transport: http(),
+}).extend(publicActionsL2());
+ 
+// eg: Metamask
+export const walletClient = createWalletClient({
+  chain: liskSepolia,
+  transport: custom(window.ethereum!),
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]:
```suggestion
  transport: custom(window.ethereum),
```

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+})
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: This code is only valid if running in the browser, I think. coz AFAIK, the `window` is available on the browser.

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+)
+```
+
+</TabItem>
+<TabItem value="client" label="client.js">
+
+```js title="client.js"
+import { createPublicClient, createWalletClient, http, custom } from 'viem'
+import { liskSepolia } from 'viem/chains'
+import { EthereumProvider } from '@walletconnect/ethereum-provider'
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx:216]: Would require the following step to be done so I would mention that!

`npm install @walletconnect/ethereum-provider`

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+      { indexed: true, name: 'to', type: 'address' },
+      {
+        indexed: true,
+        name: 'tokenId',
+        type: 'uint256',
+      },
+    ],
+  },
+  ...
+] as const;
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: This abi is incomplete as it shows syntactical errors on VS CODE! I suggest we use the abi of the hello Contract. I tried it and worked for me and the relevant code did fetch the contract! However, I got an error as described below!

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+ 
+// 1. Create contract instance
+const contract = getContract({
+  address: 'CONTRACT_ADDRESS',
+  abi: wagmiAbi,
+  // 1a. Insert a single client
+  client: publicClient,
+  // 1b. Or public and/or wallet clients
+  client: { public: publicClient, wallet: walletClient }
+})
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: I was able to fetch the contract via the following code!
```suggestion
import { publicClient, walletClientWC } from './client.js'

// 1. Create contract instance
const contract = getContract({
    address: '0xb18eb752813c2fbedfdf2be6e5e842a85a3b8539',
    abi: wagmiAbi,
    // 1a. Insert a single client
    client: publicClient,
    // 1b. Or public and/or wallet clients
    client: { public: publicClient, wallet: walletClientWC }
})
```

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+```javascript title="write-example.js"
+import { walletClientL1 } from './wallet-client.js'
+
+const hash = await walletClientL1.sendTransaction({ 
+  to: '0x...',
+  value: 1000000000000000000n
+})
+
+console.log(hash);
```
Comment (TalhaMaliktz) [inline docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: I get the following error when I try by simply using the `walletClientWC` client to run this file. I use the `node wallet-client.js` command.
```
Error: To use QR modal, please install the @walletconnect/modal package
    at I.initialize (/Users/iamtalha/Lightcurve-Code/L2/interact-blockchain/node_modules/@walletconnect/ethereum-provider/dist/index.cjs.js:1:7377)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async I.init (/Users/iamtalha/Lightcurve-Code/L2/interact-blockchain/node_modules/@walletconnect/ethereum-provider/dist/index.cjs.js:1:2743)
    at async file:///Users/iamtalha/Lightcurve-Code/L2/interact-blockchain/viem/wallet-client.js:28:18

Node.js v18.17.0
```

**Review by TalhaMaliktz — CHANGES_REQUESTED** [review]
Please consider the following!

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx (hunk 6 above)
Comment (Tschakki) [reply docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: this example only works together with walletconnect

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx (hunk 3 above)
Comment (Tschakki) [reply docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx:216]: it is only an example for projects using walletconnect. So they should have this installed already

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx (hunk 4 above)
Comment (Tschakki) [reply docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: it is just an example for an ABI, shortened for readability. People should paste their specific ABI here.

File: docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx (hunk 5 above)
Comment (Tschakki) [reply docs/building-on-lisk/interacting-with-the-blockchain/viem.mdx]: why change to `walletClientWC`?

**Review by TalhaMaliktz — APPROVED** [review]
Thank you! LGTM 🎉

**Pull request merged**